src/main/java/net/mcreator/workspace/references/ReferencesFinder.java

```diff
@@ -0,0 +1,291 @@
+		List<ModElement> elements = new ArrayList<>();
+
+		String query = new DataListEntry.Custom(element).getName();
+		workspace.getModElements().parallelStream().filter(me -> !me.equals(element)).forEach(me -> {
+			GeneratableElement ge = me.getGeneratableElement();
+			if (anyValueMatches(ge, String.class, e -> e.isAnnotationPresent(ElementReference.class), (a, t) -> {
+				ElementReference ref = a.getAnnotation(ElementReference.class);
+				return !List.of(ref.defaultValues()).contains(t) && query.equals(ref.customPrefix() + t);
+			})) {
+				elements.add(me);
+			} else if (anyValueMatches(ge, MappableElement.class, e -> true,
+					(a, t) -> t.getUnmappedValue().equals(query))) {
+				elements.add(me);
+			} else if (anyValueMatches(ge, Procedure.class, e -> true,
+					(a, t) -> t.getName() != null && !t.getName().isEmpty() && !t.getName().equals("null")
+							&& element.getName().equals(t.getName()))) {
+				elements.add(me);
+			} else if (anyValueMatches(ge, String.class, e -> e.isAnnotationPresent(BlocklyXML.class),
+					(a, t) -> t.contains(query) || t.contains(element.getName()))) {
+				elements.add(me);
+			}
+		});
+
+		return elements;
+	}
+
+	public static List<ModElement> searchTextureUsages(Workspace workspace, File texture, TextureType type) {
+		List<ModElement> elements = new ArrayList<>();
+
+		workspace.getModElements().parallelStream().forEach(me -> {
+			if (anyValueMatches(me.getGeneratableElement(), String.class, e -> {
+				TextureReference ref = e.getAnnotation(TextureReference.class);
+				return ref != null && ref.value() == type;
+			}, (a, t) -> {
+				TextureReference ref = a.getAnnotation(TextureReference.class);
+				return !List.of(ref.defaultValues()).contains(t) && workspace.getFolderManager()
+						.getTextureFile(FilenameUtilsPatched.removeExtension(t), type).equals(texture);
+			})) {
+				elements.add(me);
+			}
+		});
+
+		return elements;
+	}
+
+	public static List<ModElement> searchModelUsages(Workspace workspace, Model model) {
+		List<ModElement> elements = new ArrayList<>();
+
+		workspace.getModElements().parallelStream().forEach(me -> {
+			if (anyValueMatches(me.getGeneratableElement(), Model.class,
+					e -> e.isAnnotationPresent(ModelReference.class),
+					(a, t) -> model.equals(t) || TexturedModel.getModelTextureMapVariations(model).contains(t))) {
+				elements.add(me);
+			}
+		});
+
+		return elements;
+	}
+
+	public static List<ModElement> searchSoundUsages(Workspace workspace, SoundElement sound) {
+		List<ModElement> elements = new ArrayList<>();
+
+		workspace.getModElements().parallelStream().forEach(me -> {
+			if (anyValueMatches(me.getGeneratableElement(), Sound.class, e -> true,
+					(a, t) -> t.getUnmappedValue().replaceFirst("CUSTOM:", "").equals(sound.getName())))
+				elements.add(me);
+		});
+
+		return elements;
+	}
+
+	public static List<ModElement> searchStructureUsages(Workspace workspace, String structure) {

```

Comment:
Structures can also be used in procedures and in the future in features

Reply:
Isn't this the current behavior?
https://github.com/MCreator/MCreator/blob/df0a48b437116d3ca346f4cd09700b9e39eaff32/src/main/java/net/mcreator/workspace/references/ReferencesFinder.java#L129

Reply:
The problem I have with this is it will also match usages in whole strings. I would at least match it as `="structureid"` or something similar, to reduce cases where there could be false matches

Reply:
I thought it may be better to include all matches rather than ignore cases like that and let user face runtime game error because they removed and element the usages dialog did not highlight. I'll see though.

Reply:
I agree, but checking if the string is in quotes would still catch all cases

Reply:
Only it can be met in corner braces rather than quotes (">" + structure + "</field>").

Reply:
:heavy_check_mark: Fixed